cli/compose/types/types.go

```diff
@@ -260,6 +262,15 @@ type ServiceSecretConfig struct {
 	Mode   *uint32
 }
 
+// ServiceConfigObjConfig is the config obj configuration for a service
+type ServiceConfigObjConfig struct {
```

Comment:
Are they that separate? At the very least they share 5 fields! Even as other fields are added, these 5 fields will still overlap.

If not this solution, how about a common struct for these fields that gets included in both?